lib/github.dart

```diff
@@ -0,0 +1,1029 @@
+// Copyright (c) 2019, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+library playground;
+
+import 'dart:async';
+import 'dart:convert' show json;
+import 'dart:html' hide Console;
+import 'dart:math';
+
+import 'package:encrypt/encrypt.dart';
+import 'package:http/http.dart' as http;
+import 'package:mdc_web/mdc_web.dart';
+import 'package:stream_transform/stream_transform.dart';
+
+import 'dart_pad.dart';
+import 'elements/analysis_results_controller.dart';
+import 'elements/button.dart';
+import 'elements/elements.dart';
+import 'playground.dart';
+import 'sharing/gists.dart';
+import 'sharing/mutable_gist.dart';
+
+class GitHubUIController {
+  static const kEntryPointGitHubOAuthInitiate = 'github_oauth_initiate';
```

Comment:
Why is this the only constant? I see a lot of duplicate /inline strings declared in this file, like `github_oauth_token`, `github_avatar_url`, `GIST_ALREADY_FORK`, etc. We should use more constants if possible.

Reply:
thanks for catching this, constants have been added (and not prefixed with 'k').